apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-preview.js

```diff
@@ -23,85 +30,80 @@ import BlockPreview from './block-preview';
+		if ( ! ref || ! ref.current ) {
+			return;
+		}
 
-	const updateTemplateTitle = () => {
-		// Get DOM reference.
 		setTimeout( () => {
-			if ( ! ref || ! ref.current ) {
+			const preview = ref.current.querySelector( '.block-editor-block-preview__content' );
+			if ( ! preview ) {
 				return;
 			}
 
-			// Try to get the preview content element.
-			const previewContainerEl = ref.current.querySelector(
-				'.block-editor-block-preview__content'
+			const previewScale = parseFloat(
+				get( preview, [ 'style', 'transform' ], '' )
+					.replace( 'scale(', '' )
+					.replace( ')', '' )
 			);
-			if ( ! previewContainerEl ) {
-				return;
+			if ( previewScale ) {
+				const titleOffset = TITLE_DEFAULT_HEIGHT * previewScale;
+				setTitleTransform( { scale: previewScale, offset: titleOffset } );
 			}
-
-			// Try to get the `transform` css rule from the preview container element.
-			const elStyles = window.getComputedStyle( previewContainerEl );
-			if ( elStyles && elStyles.transform ) {
-				const titleElement = ref.current.querySelector( '.editor-post-title' );
-				if ( titleElement ) {
-					// Apply the same transform css rule at template title element.
-					titleElement.style.transform = elStyles.transform;
-				}
-
-				// Pick up scale factor from `transform` css.
-				let scale = elStyles.transform.replace( /matrix\((.+)\)$/i, '$1' ).split( ',' );
-				scale = scale && scale.length ? Number( scale[ 0 ] ) : null;
-				scale = isNaN( scale ) ? null : scale;
-
-				// Try to adjust vertical offset of the large preview.
-				const offsetCorrectionEl = previewContainerEl.closest(
-					'.template-selector-preview__offset-correction'
-				);
-
-				if ( offsetCorrectionEl && scale ) {
-					const titleHeight = titleElement ? titleElement.offsetHeight : null;
-					offsetCorrectionEl.style.top = `${ titleHeight * scale }px`;
-				}
-			}
-
-			setVisibility( 'visible' );
-		}, 300 );
+		}, 500 );
 	};
 
+	const updatePreviewViewport = useCallback( () => {
+		if ( ! ref || ! ref.current ) {
+			return;
+		}
+		const wrapperWidth = ref.current.clientWidth;
+		if ( wrapperWidth >= viewportWidth ) {
+			setPreviewViewport( wrapperWidth );
+		} else {
+			setPreviewViewport( viewportWidth );
+		}
+	}, [ viewportWidth ] );
+
 	useLayoutEffect( () => {
-		setVisibility( 'hidden' );
-		updateTemplateTitle();
-	}, [ blocks ] );
+		updatePreviewViewport();
+		updatePreviewTitle();
+	}, [ blocks, updatePreviewViewport ] );
 
 	useEffect( () => {
 		if ( ! blocks || ! blocks.length ) {
 			return;
 		}
 
 		const rePreviewTemplate = () => {
-			updateTemplateTitle();
+			updatePreviewViewport();
+			updatePreviewTitle();
 			triggerRecompute();
 		};
 
 		const refreshPreview = debounce( rePreviewTemplate, THRESHOLD_RESIZE );
 		window.addEventListener( 'resize', refreshPreview );
 
+		// In wp-admin, listen to the jQuery `wp-collapse-menu` event to refresh the preview on sidebar toggle.
+		if ( window.jQuery ) {
```

Comment:
😭

Reply:
Yeah, TIL jQuery events are not compatible with vanilla (the opposite is true though: of course jQuery can listen to vanilla-defined events).
And indeed [wp-admin triggers the sidebar event through jQuery](https://github.com/WordPress/WordPress/blob/f8964463c112b7d86e3bfd058a67145dc3966e9b/wp-admin/js/common.js#L546), so there's no way around it. ☹️ 

I think this is a neat enough solution though.
We aren't importing jQuery so we don't bloat the build, and just rely on wp-admin that puts jQuery into global.